src/commands/schema_diff.ts

```diff
@@ -0,0 +1,108 @@
+import yargs from 'yargs';
+import { fillSingleProject } from '../utils/enrichers.js';
+import { ProjectScopeProps } from '../types';
+import { createPatch } from 'diff';
+import { Database } from '@neondatabase/api-client';
+import chalk from 'chalk';
+import { writer } from '../writer.js';
+
+type SchemaDiffProps = ProjectScopeProps & {
+  baseBranch: string;
+  compareBranch: string;
+  database: string;
+};
+
+const COLORS = {
+  added: chalk.green,
+  removed: chalk.red,
+  header: chalk.yellow,
+  section: chalk.magenta,
+};
+
+type ColorId = keyof typeof COLORS;
+
+export const command = 'schema-diff';
+export const describe = 'Compare schemas of two branches';
+export const builder = (argv: yargs.Argv) =>
+  argv
+    .option({
+      'project-id': {
+        describe: 'Project Id',
+        type: 'string',
+      },
+      'base-branch': {
+        describe: 'Base Branch Id',
+        type: 'string',
+      },
+      'compare-branch': {
+        describe: 'Compare Branch Id',
+        type: 'string',
+      },
+      database: {
+        describe: 'Database name',
+        type: 'string',
+      },
+    })
+    .middleware(fillSingleProject as any);
+
+export const handler = (args: yargs.Argv) => {
+  return schemaDiff(args as any);
+};
+
+export const schemaDiff = async (props: SchemaDiffProps) => {
+  const baseBranch = props.baseBranch;
+  const compareBranch = props.compareBranch;
+
+  const [baseDatabase, compareDatabase] = await Promise.all([
+    fetchDatabase(baseBranch, props),
+    fetchDatabase(compareBranch, props),
+  ]);
+
+  const [baseSchema, compareSchema] = await Promise.all([
+    fetchSchema(baseBranch, baseDatabase, props),
+    fetchSchema(compareBranch, compareDatabase, props),
+  ]);
+
+  const patch = createPatch(
+    props.database,
+    baseSchema,
+    compareSchema,
+    `Base Branch: ${baseBranch}`,
+    `Compare Branch: ${compareBranch}`,
+  );
+  writer(props).text(colorize(patch));
```

Comment:
is there an option to support json/yaml output? Maybe library can return something like an array of diffs?